Unstable test `EmbeddedSubProcessTest.shouldNotTriggerBoundaryEventWhenFlowscopeIsInterrupted`

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? At least three times
- Does it block your work? It's beginning to block merges
- Do we suspect that it is a real failure? Probably not

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
java.lang.AssertionError: 

Expecting actual:
  []
to contain exactly (and in same order):
  ["msgBoundary", "eventSubProcessStartEvent"]
but could not find the following elements:
  ["msgBoundary", "eventSubProcessStartEvent"]

	at io.camunda.zeebe.engine.processing.bpmn.subprocess.EmbeddedSubProcessTest.shouldNotTriggerBoundaryEventWhenFlowscopeIsInterrupted(EmbeddedSubProcessTest.java:663)
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

**Logs**
https://github.com/camunda/zeebe/actions/runs/4289453228/jobs/7472507432
https://github.com/camunda/zeebe/actions/runs/4290687672/jobs/7475272618
https://github.com/camunda/zeebe/actions/runs/4293686871/jobs/7481685355

--- a/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/EmbeddedSubProcessConcurrencyTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/subprocess/EmbeddedSubProcessConcurrencyTest.java
@@ -18,6 +18,7 @@ import io.camunda.zeebe.model.bpmn.builder.EventSubProcessBuilder;
 import io.camunda.zeebe.protocol.impl.record.value.message.MessageRecord;
 import io.camunda.zeebe.protocol.record.ValueType;
 import io.camunda.zeebe.protocol.record.intent.MessageIntent;
+import io.camunda.zeebe.protocol.record.intent.MessageSubscriptionIntent;
 import io.camunda.zeebe.protocol.record.intent.ProcessEventIntent;
 import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
 import io.camunda.zeebe.protocol.record.value.BpmnElementType;
@@ -95,10 +96,15 @@ public class EmbeddedSubProcessConcurrencyTest {
             .withVariable("correlationKey", "correlationKey")
             .create();
 
-    RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
-        .withProcessInstanceKey(processInstanceKey)
-        .withElementId("task")
-        .await();
+    assertThat(
+            RecordingExporter.messageSubscriptionRecords(MessageSubscriptionIntent.CREATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .limit(2))
+        .describedAs(
+            "The 2 message subscriptions must be created before we publish the "
+                + "messages. As the messages have a TTL of 0 seconds")
+        .describedAs("")
+        .hasSize(2);
 
     // when
     // We need to make sure no records are written in between the publish commands. This could

